### PR TITLE
test(i): Test top level agg. wt compound relational filter

### LINKED ISSUE #1868: Add tests for aggregates with compound relational filters

### PR DESCRIPTION
**Pull request by AndrewSisley**
## Relevant issue(s)

Resolves #1868

## Description

Adds a test for aggregates with a compound relational filter.

Documents https://github.com/sourcenetwork/defradb/issues/1869

I believe this predates the release, and may be a fiddly fix with lots more testing required, so I'm documenting it now and moving on - maybe it will be fixed before the release goes out, but will see.

### CODE REVIEW

**Comment by AndrewSisley** [comment]
`AssertPanicAndSkipChangeDetection` does not work with the http client, the test will not pass until sorted.  Looking at how to handle that now.

File: tests/integration/utils2.go [hunk 1]
```diff
@@ -196,17 +196,24 @@ func init() {
 	}
 }
 
-// AssertPanicAndSkipChangeDetection asserts that the code of function actually panics,
+// AssertPanic asserts that the code inside the specified PanicTestFunc panics.
```
Comment (shahzadlone) [inline tests/integration/utils2.go:199]: nitpick(non-blocking): Rename suggestion
```suggestion
// AssertPanicOrSkip asserts that the code inside the specified PanicTestFunc panics.
```

**Review by shahzadlone — APPROVED** [review]
LGTM. Just a non-blocking name suggestion.

File: tests/integration/utils2.go (hunk 1 above)
Comment (AndrewSisley) [reply tests/integration/utils2.go:199]: :) I do disagree here, although I think my views over time have shifted, and I might have been the one to originally ask for the longer name.  Will leave as `AssertPanic` unless anyone else requests the change before I rebase and merge.

**Pull request merged**